__tests__/app.test.js

```diff
@@ -103,4 +104,47 @@ describe('GET /api', () => {
   });
 
   
+
+  describe('GET /api/articles/:article_id/comments', () => {
+    test('200: responds with an array of comments for the given article_id', () => {  
+      return request(app)
+        .get(`/api/articles/1/comments`)
+        .expect(200)
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments).toBeInstanceOf(Array);
+          expect(comments.length).toBeGreaterThan(0);
   
+          for (const comment of comments) {
+            expect(comment).toMatchObject({
+              comment_id: expect.any(Number),
+              votes: expect.any(Number),
+              created_at: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              article_id: 1,
+            });
+          }
+          const timestamps = comments.map(comment => new Date(comment.created_at).getTime());
+        expect(timestamps).toEqual([...timestamps].sort((a, b) => b - a));
+        });
+    })
+
+    test('GET:404 sends an appropriate status and error message when given a valid but non-existent id', () => {
+        return request(app)
+          .get('/api/articles/100/comments')
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe('comment does not exist');
+          });
+      });
+
+    test('GET:400 sends an appropriate status and error message when given an invalid id', () => {
+        return request(app)
+        .get('/api/articles/not-an-article/comments')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Bad Request')
+        })
+    });
+})
```

Comment:
Missing one more test, need a 200 test to check that when a request is made for an article with no comments, an empty array is returned.

Reply:
alright. doing so now

also just about sending a pull request for task 7 😊

Reply:
I did not realise this checking for correct respons will take my time. thanks katherine. Going back to solve this was really a eye opening.

Reply:
submitting a fresh nchelp pr now